test/unit/account_test.rb

```diff
@@ -750,64 +750,27 @@ def test_default_service_id
-
-    master_account.signup_provider(master_account.application_plans.first) do |provider, user|
-      @provider, @user = provider, user
-      provider.subdomain = "foo"
-      provider.org_name = "bar"
-      provider.sample_data = false
-      user.password = user.password_confirmation = "foobar"
-      user.email = "foo@example.com"
-    end
+    master_plan = master_account.default_application_plans.first!
 
-    @user.activate
-    @provider.create_sample_data!
+    provider = FactoryBot.create(:simple_provider, provider_account: master_account)
+    cinstance_tenant = FactoryBot.create(:cinstance, plan: master_plan, user_account: provider)
 
-    cis = @provider.default_service.cinstances.map(&:id)
-    aps = @provider.default_service.application_plans.map(&:id)
+    service = FactoryBot.create(:simple_service, account: provider)
+    buyer = FactoryBot.create(:simple_buyer, provider_account: provider)
+    tenant_plan = FactoryBot.create(:application_plan, issuer: service)
+    cinstance_buyer = FactoryBot.create(:cinstance, plan: tenant_plan, user_account: buyer)
 
-    @provider.expects(:destroy_all_contracts)
+    assert_not_equal 0, provider.provided_cinstances.reload.count
+    assert_not_equal 0, provider.provided_plans.reload.count
+    assert_not_equal 0, provider.bought_plans.reload.count
 
-    @provider.destroy
-    assert_equal 0, Cinstance.where(id: cis).count
-    assert_equal 0, ApplicationPlan.where(id: aps).count
-  end
+    provider.destroy!
 
-  test '#signup provider with api_as_product' do
-    master_account.account_plans.default!(master_account.account_plans.first)
-    master_account.signup_provider(master_account.application_plans.first) do |provider, user|
-      provider.stubs(:provider_can_use?).returns(false)
-      provider.expects(:provider_can_use?).with(:api_as_product).returns(true).at_least_once
-      @provider, @user = provider, user
-      provider.subdomain = "foo"
-      provider.org_name = "bar"
-      provider.sample_data = false
-      user.password = user.password_confirmation = "foobar"
-      user.email = "foo@example.com"
-    end
-    assert_equal BackendApi.default_api_backend, @provider.first_service.api_backend
-  end
-
-
-  test '#signup provider without api_as_product' do
-    master_account.account_plans.default!(master_account.account_plans.first)
-    master_account.signup_provider(master_account.application_plans.first) do |provider, user|
-      @provider, @user = provider, user
-      provider.stubs(:provider_can_use?).returns(false)
-      provider.expects(:provider_can_use?).with(:api_as_product).returns(false).at_least_once
-      provider.subdomain = "foo"
-      provider.org_name = "bar"
-      provider.sample_data = false
-      user.password = user.password_confirmation = "foobar"
-      user.email = "foo@example.com"
-    end
-    assert_equal BackendApi.default_api_backend, @provider.first_service.api_backend
```

Comment:
I remember discussing about this when these couple tests [were first introduced](https://github.com/3scale/porta/pull/1211). Back then the idea was having a way to test that with or without APIAP a default API backend would be always created on signup. IMO, we still need to test for that somehow.

Reply:
We do @guicassolato 
https://github.com/3scale/porta/blob/7c6b46b84bc40dd06e461b33e54cb6d517a303fd/test/unit/signup/account_manager_test.rb#L130-L153

The problem is we were doing the signup through 2 ways, one was the signup_provider (which is what this was testing), and another was the AccountManager which is the only one staying now and the one being tested in what I pasted above.

This is what actually does it:
https://github.com/3scale/porta/blob/7c6b46b84bc40dd06e461b33e54cb6d517a303fd/app/lib/signup/provider_account_manager.rb#L36-L38

Reply:
Perfect!